src/internal/m365/collection/drive/limiter_test.go

```diff
@@ -54,36 +51,18 @@ func TestLimiterUnitSuite(t *testing.T) {
 	suite.Run(t, &LimiterUnitSuite{Suite: tester.NewUnitSuite(t)})
 }
 
-// TestGet_PreviewLimits checks that the limits set for preview backups in
-// control.Options.ItemLimits are respected. These tests run a reduced set of
-// checks that don't examine metadata, collection states, etc. They really just
-// check the expected items appear.
-func (suite *LimiterUnitSuite) TestGet_PreviewLimits() {
-	metadataPath, err := path.BuildMetadata(
-		tenant,
-		user,
-		path.OneDriveService,
-		path.FilesCategory,
-		false)
-	require.NoError(suite.T(), err, "making metadata path", clues.ToCore(err))
-
-	drive1 := models.NewDrive()
-	drive1.SetId(ptr.To(idx(drive, 1)))
-	drive1.SetName(ptr.To(namex(drive, 1)))
-
-	drive2 := models.NewDrive()
-	drive2.SetId(ptr.To(idx(drive, 2)))
-	drive2.SetName(ptr.To(namex(drive, 2)))
+type backupLimitTest struct {
+	name       string
+	limits     control.PreviewItemLimits
+	drives     []models.Driveable
+	enumerator mock.EnumerateItemsDeltaByDrive
+	// Collection name -> set of item IDs. We can't check item data because
+	// that's not mocked out. Metadata is checked separately.
+	expectedCollections map[string][]string
+}
 
-	table := []struct {
-		name       string
-		limits     control.PreviewItemLimits
-		drives     []models.Driveable
-		enumerator mock.EnumerateItemsDeltaByDrive
-		// Collection name -> set of item IDs. We can't check item data because
-		// that's not mocked out. Metadata is checked separately.
-		expectedCollections map[string][]string
-	}{
+func backupLimitTable(drive1, drive2 models.Driveable) []backupLimitTest {
```

Comment:
nit: Passing these values in makes the tests fragile. These values are directly returned via the `drives` struct field, but the output of the enumerator builds the drive ID manually. This means that if any of the passed in drives doesn't follow the expected drive ID format things could break. It would probably be better to just construct the drives in this function instead of taking them as parameters

Reply:
fair, will do.